runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -0,0 +1,1376 @@
+                                                 TR::ILOpCodes opcode)
+   {
+   TR::Compilation *comp = opt->comp();
+
+   anchorOldChildren(opt, treeTop, node);
+   node->setAndIncChild(0, firstChild);
+   node->setAndIncChild(1, secondChild);
+   node->setNumChildren(2);
+   
+   if (mode == doScalarization)
+      {
+      int32_t elementSize = OMR::DataType::getSize(elementType);
+      int numLanes = vectorLength/8/elementSize;
+
+      if (firstChild->getOpCodeValue() == TR::aload) aloadHandler(opt, treeTop, firstChild, elementType, vectorLength, mode);
+      if (secondChild->getOpCodeValue() == TR::aload) aloadHandler(opt, treeTop, secondChild, elementType, vectorLength, mode);
+
+      TR::Node::recreate(node, opcode);
+
+      for (int i = 1; i < numLanes; i++)
+         {
+         TR::Node *newAddNode = TR::Node::create(node, opcode, 2);
+         addScalarNode(opt, node, numLanes, i, newAddNode);
+         newAddNode->setAndIncChild(0, getScalarNode(opt, firstChild, i));
+         newAddNode->setAndIncChild(1, getScalarNode(opt, secondChild, i));
+         }
+      }
+   else if (mode == doVectorization)
+      {
+      TR::DataType vectorType = OMR::DataType(elementType).scalarToVector();
+
+      if (firstChild->getOpCodeValue() == TR::aload) vectorizeLoadOrStore(opt, firstChild, vectorType);
+      if (secondChild->getOpCodeValue() == TR::aload) vectorizeLoadOrStore(opt, secondChild, vectorType);
+
+      TR::Node::recreate(node, TR::vcall);
+      }
+      
+   return node;
+}
+
+
+// high level methods are disabled because they need excpetion handling to be inserted
+TR_VectorAPIExpansion::methodTableEntry
+TR_VectorAPIExpansion::methodTable[] =
+   {
+   {loadIntrinsicHandler,  TR::NoType, Vector,  {Unknown, elementType, numLanes}},                           // jdk_internal_vm_vector_VectorSupport_load
+   {storeIntrinsicHandler, TR::NoType, Unknown, {Unknown, elementType, numLanes, Unknown, Unknown, Vector}}, // jdk_internal_vm_vector_VectorSupport_store
+   {binaryIntrinsicHandler,TR::NoType, Vector,  {Unknown, Unknown, elementType, numLanes, Vector, Vector}},  // jdk_internal_vm_vector_VectorSupport_binaryOp
+   {unsupportedHandler /*fromArrayHandler*/,      TR::Float,  Vector,  {Species}}, // jdk_incubator_vector_FloatVector_fromArray,
+   {unsupportedHandler /*intoArrayHandler*/,      TR::Float,  Unknown, {Vector}},  // jdk_incubator_vector_FloatVector_intoArray,
+   {unsupportedHandler,    TR::Float,  Vector,  {Species}},                 // jdk_incubator_vector_FloatVector_fromArray_mask
+   {unsupportedHandler,    TR::Float,  Unknown, {}},                        // jdk_incubator_vector_FloatVector_intoArray_mask
+   {addHandler,            TR::Float,  Vector,  {Vector, Vector}},          // jdk_incubator_vector_FloatVector_add
+   {unsupportedHandler,    TR::Float,  Vector,  {}}                         // jdk_incubator_vector_VectorSpecies_indexInRange
+   };
```

Comment:
This seems reliant on `Last_vector_api_method - First_vector_api_method` being a constant value at compile time. I think this code breaks if someone were to insert a new Vector API method without knowing they need to update this table.

We can fix this via `static_assert` ensuring that the number of elements here is exactly `Last_vector_api_method - First_vector_api_method`.

Reply:
Yeah, meant to do that and somehow forgot. Thanks for noticing!

Reply:
And will try the J2SE_V17 check. Thanks!

Reply:
Just pushed both